File: clients/python/queueti/_options.py
from dataclasses import dataclass, field
from typing import Awaitable, Callable

TokenRefresher = Callable[[], Awaitable[str]]


@dataclass
class ConnectOptions:
    token: str | None = None
    token_refresher: TokenRefresher | None = None
    insecure: bool = False


@dataclass
class PublishOptions:
    metadata: dict[str, str] = field(default_factory=dict)


@dataclass
class ConsumerOptions:
    concurrency: int = 1
    visibility_timeout_seconds: int | None = None
    consumer_group: str = ""

    def __post_init__(self) -> None:
        if self.concurrency < 1:
            raise ValueError(f"concurrency must be >= 1, got {self.concurrency}")


@dataclass
class BatchOptions:
    batch_size: int
    visibility_timeout_seconds: int | None = None
    consumer_group: str = ""

    def __post_init__(self) -> None:
        if self.batch_size < 1:
            raise ValueError(f"batch_size must be >= 1, got {self.batch_size}")

File: clients/node/src/options.ts
export type TokenRefresher = () => Promise<string>

export interface TLSOptions {
  rootCerts?: Buffer      // PEM-encoded CA certificate(s); uses system CAs when omitted
  privateKey?: Buffer     // PEM-encoded client private key for mTLS; requires certChain
  certChain?: Buffer      // PEM-encoded client certificate chain for mTLS; requires privateKey
  serverNameOverride?: string  // override the hostname used for TLS SNI/verification
}

export interface ConnectOptions {
  insecure?: boolean
  tls?: TLSOptions        // custom TLS config; ignored when insecure is true
  token?: string
  tokenRefresher?: TokenRefresher
}

export interface PublishOptions {
  metadata?: Record<string, string>
  key?: string
}

export interface ConsumerOptions {
  concurrency?: number
  visibilityTimeoutSeconds?: number
  signal?: AbortSignal
  consumerGroup?: string
}

export interface BatchOptions {
  batchSize: number
  visibilityTimeoutSeconds?: number
  consumerGroup?: string
}
